shared/protocol/src/main/java/io/pravega/shared/protocol/netty/AppendDecoder.java

```diff
@@ -163,22 +205,15 @@ private ByteBuf getAppendDataBuf(WireCommands.AppendBlockEnd blockEnd, int sizeO
                 // Work around a bug in netty:
                 // See https://github.com/netty/netty/issues/5597
                 if (appendDataBuf.readableBytes() == 0) {
-                    appendDataBuf.release();
+                    currentBlock.release();
                     appendDataBuf = wrappedBuffer(((WireCommands.PartialEvent) cmd).getData(), blockEnd.getData());
                 } else {
                     appendDataBuf = wrappedBuffer(appendDataBuf, ((WireCommands.PartialEvent) cmd).getData(), blockEnd.getData());
                 }
             }
         }
 
-        // Make a copy of the ByteBuf as the readable bytes of the result may be significantly less than the total allocated
```

Comment:
See this comment for why this copy was added. We need to test to confirm this will not cause problemms.

Reply:
@RaulGracia is testing this with unthrottled loads. If we don't get any memory problems we should be OK. 

Since we've added that extra copy we have made numerous improvements in the Segment Stores' throttler and it is now a lot more aggressive in keeping incoming data flow under control. My expectation is that this should no longer be an issue and I want to see if it actually serves any purpose.

In addition, there is no longer a heap issue, as none of the source buffers are stored in the heap (the changes I made in WireCommands eliminated the heap buffer allocations). This will now point to sliced direct mem buffers from Netty.

Reply:
We executed several perf tests and had no issues.